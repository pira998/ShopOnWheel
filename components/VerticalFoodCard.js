import React from "react";

import { View,TouchableOpacity,Image,Text } from "react-native";

import { COLORS,FONTS,SIZES,icons,images } from "../constants";



const VerticalFoodCard =({containerStyle,item,onPress,navigation})=>{
    return(
        <TouchableOpacity
        style={{
            width:200,
            padding:SIZES.radius,
            alignItems:'center',
            borderRadius:SIZES.radius,
            backgroundColor:COLORS.lightGray2,
            ...containerStyle
        }}
        onPress={onPress}
        >

            {/* calories and Favourits*/}
            <View style={{flexDirection:'row'}}>
                {/* calories  */}
                <View style={{flex:1,flexDirection:'row'}}>
                    <Image
                    source={icons.calories}
                    style={{
                        width:30,
                        height:30
                    }}
                    />
                    <Text style={{color:COLORS.darkGray}}>{item.calories} calories</Text>
                </View>
                {/* favouritss  */}
                <Image
                source={icons.love}
                style={{width:20,height:20,tintColor:item.rating>4.5?COLORS.primary :COLORS.gray}}/>

            </View>

            {/* image  */}

            <View style={{
                height:150,
                width:150,
                alignItems:'center',
                justifyContent:'center',
                borderRadius:75,
                overflow:'hidden',
              
                marginBottom:19
            }}>
                <Image
                source={item.photo}
                style={{height:'100%',width:'100%'}}
                />
            </View>

            {/* info */}
            <View
            style={{
                alignItems:'center',
                marginTop:-20
            }}>

                <Text style={{...FONTS.h3}}>{item.name}</Text>
                {/* <Text style={{color:COLORS.darkGray2,textAlign:'center',...FONTS.body4}}>{item.description}</Text> */}
                <Text style={{marginTop:SIZES.radius,...FONTS.h3}}>{item.duration}</Text>
            </View>


            
        </TouchableOpacity>
    )
}

export default VerticalFoodCard;