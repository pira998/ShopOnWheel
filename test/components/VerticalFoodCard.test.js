import React from 'react';
import renderer from 'react-test-renderer';
import VerticalFoodCard from '../../components/VerticalFoodCard'


test('renders correctly', () => {
  const item = {
      name:'testing',
      image:'testImage',
      calories:0
  }
  const tree = renderer.create(<VerticalFoodCard item={item} />).toJSON();
  expect(tree).toMatchSnapshot();
});