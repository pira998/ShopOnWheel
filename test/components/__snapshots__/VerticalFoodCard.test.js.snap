// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#F5F5F8",
      "borderRadius": 12,
      "opacity": 1,
      "padding": 12,
      "width": 200,
    }
  }
>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
        }
      }
    >
      <Image
        source={1}
        style={
          Object {
            "height": 30,
            "width": 30,
          }
        }
      />
      <Text
        style={
          Object {
            "color": "#525C67",
          }
        }
      >
        0
         calories
      </Text>
    </View>
    <Image
      source={1}
      style={
        Object {
          "height": 20,
          "tintColor": "#898B9A",
          "width": 20,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderRadius": 75,
        "height": 150,
        "justifyContent": "center",
        "marginBottom": 19,
        "overflow": "hidden",
        "width": 150,
      }
    }
  >
    <Image
      style={
        Object {
          "height": "100%",
          "width": "100%",
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "marginTop": -20,
      }
    }
  >
    <Text
      style={
        Object {
          "fontFamily": "sans-serif",
          "fontSize": 16,
          "lineHeight": 22,
        }
      }
    >
      testing
    </Text>
    <Text
      style={
        Object {
          "fontFamily": "sans-serif",
          "fontSize": 16,
          "lineHeight": 22,
          "marginTop": 12,
        }
      }
    />
  </View>
</View>
`;
